.github/workflows/osx_win_python_builds.yml

```diff
@@ -0,0 +1,393 @@
+
+      - name: Build Valhalla
+        run: |
+          cmake --build build --config Release -- /clp:ErrorsOnly /p:BuildInParallel=true /m:4
+          cmake --build build --config Release --target install
+
+      - name: Test Executable
+        shell: bash
+        run: |
+          set PATH=$PATH:${{ github.workspace }}/build/vcpkg_installed/$BUILD_TYPE/bin
+          ./build/$BUILD_TYPE/valhalla_build_tiles.exe -c ./test/win/valhalla.json ./test/data/utrecht_netherlands.osm.pbf
+          ./build/$BUILD_TYPE/valhalla_run_isochrone.exe --config ./test/win/valhalla.json -j "{\"locations\": [{\"lat\": 52.10205, \"lon\": 5.114651}], \"costing\": \"auto\", \"contours\":[{\"time\":15,\"color\":\"ff0000\"}]}"
+          ./build/$BUILD_TYPE/valhalla_service.exe ./test/win/valhalla.json isochrone "{\"locations\": [{\"lat\": 52.10205, \"lon\": 5.114651}], \"costing\": \"auto\", \"contours\":[{\"time\":15,\"color\":\"ff0000\"}]}"
+      
+      - name: Determine package to publish
+        if: github.event_name != 'pull_request' && ${{ github.repository_owner == 'valhalla' }}
+        shell: bash
+        run: |
+          VALHALLA_RELEASE_PKG="pyvalhalla"
+          if [[ "${GITHUB_REF#refs/heads/}" != "refs/tags*" ]]; then
+            VALHALLA_RELEASE_PKG="${VALHALLA_RELEASE_PKG}-git"
+          fi
+          echo "VALHALLA_RELEASE_PKG=${VALHALLA_RELEASE_PKG}" >> $GITHUB_ENV
+        
+      - name: Print libs
+        shell: bash
+        run: |
+          ls -l build/vcpkg_installed/custom-x64-windows/lib
+          ls -l build/vcpkg_installed/custom-x64-windows/bin
+      
+      - uses: pypa/cibuildwheel@v2.23.3
+        if: ${{ (github.event_name != 'pull_request') && (github.repository_owner == 'valhalla') }}
+        env:
+          # used in setup.py
+          CIBW_ENVIRONMENT: >
+            VALHALLA_RELEASE_PKG=${{ env.VALHALLA_RELEASE_PKG }}
+          CIBW_ENVIRONMENT_WINDOWS: >
+            VCPKG_ARCH_ROOT=build/vcpkg_installed/custom-x64-windows
+          CIBW_REPAIR_WHEEL_COMMAND_WINDOWS: >
+            delvewheel repair --add-path build/vcpkg_installed/custom-x64-windows/bin {wheel} -w {dest_dir}
+
+      - name: Upload wheels
+        if: ${{ (github.event_name != 'pull_request') && (github.repository_owner == 'valhalla') }}
+        uses: actions/upload-artifact@v4
+        with:
+          name: wheels-win-amd64
+          path: wheelhouse/*.whl
+
+      - name: Setup tmate session
+        uses: mxschmitt/action-tmate@v3
+        # only run this if manually invoked or a previous job failed
+        if: ${{ (github.event_name != 'pull_request') && (github.repository_owner == 'valhalla') }}
+        with:
+          detached: true
+
+  verify_wheels:
+    name: Verify wheels with executables in a fresh environment
+    needs: [build_python_wheels_linux, build_osx, build_win]
+    runs-on: python:slim-bullseye
+    if: ${{ (github.event_name != 'pull_request') && (github.repository_owner == 'valhalla') }}
```

Comment:
since I had so much trouble with linking dependencies on linux, I added a verification step to make mostly sure the bindings are really working as intended before we upload them to PyPI. the same test is being run when the wheels are produced, but the system it's running on has lots of system dependencies still installed like `libexpat` which we can't uninstall bcs python links to it. anyways, this runs in a fresh environment.